File: generics/Demo1.java
import java.util.ArrayList;
import java.util.List;

class Container<T> {
    private List<T> items = new ArrayList<>();

    public void addItem(T item) {
        items.add(item);
    }

    public T getItem(int index) {
        return items.get(index);
    }

    public void print() {
        for (T item : items) {
            System.out.print(item + " ");
        }

        System.out.println();
    }
}

class Demo1 {
    public static void main(String[] args) {
        Container<String> cont1 = new Container<String>();

        cont1.addItem("one");
        cont1.addItem("two");
        cont1.addItem("three");

        System.out.println("Item at index 2 is " + cont1.getItem(2));

        cont1.print();

        Container<Integer> cont2 = new Container<Integer>();

        cont2.addItem(1);
        cont2.addItem(2);
        cont2.addItem(3);

        System.out.println("Item at index 2 is " + cont2.getItem(2));

        cont2.print();
    }
}

File: generics/Demo1.kt
class Container<T> {
    val items = mutableListOf<T>()

    fun addItem(item: T) {
        items.add(item)
    }

    fun getItem(index: Int): T {
        return items[index]
    }

    fun print() {
        items.forEach { item -> 
            print("$item ")
        }

        println()
    }
}

fun main() {
    val cont1 = Container<String>()

    cont1.addItem("one")
    cont1.addItem("two")
    cont1.addItem("three")

    println("Item at index 2 is ${cont1.getItem(2)}")

    cont1.print()

    val cont2 = Container<Int>()

    cont2.addItem(1)
    cont2.addItem(2)
    cont2.addItem(3)

    println("Item at index 2 is ${cont2.getItem(2)}")

    cont2.print()

}
